packages/web/src/components/createCell.tsx

```diff
@@ -124,25 +159,34 @@ export function createCell<CellProps = any>({
         }
         {...beforeQuery(variables)}
       >
-        {({ error, loading, data, ...queryRest }) => {
-          if (error) {
-            if (Failure) {
-              return <Failure error={error} {...queryRest} {...props} />
-            } else {
-              throw error
-            }
-          } else if (loading) {
-            return <Loading {...queryRest} {...props} />
-          } else if (data) {
-            if (typeof Empty !== 'undefined' && isEmpty(data)) {
-              return <Empty {...queryRest} {...props} />
-            } else {
+        {({ status, error, data, ...queryRest }) => {
+          switch (status) {
+            case Status.Error:
+              if (Failure) {
+                return <Failure error={error} {...queryRest} {...props} />
+              } else {
+                throw error
+              }
+            case Status.Loading:
+              return <Loading {...queryRest} {...props} />
+            case Status.Empty:
+              if (typeof Empty !== 'undefined') {
+                return <Empty {...queryRest} {...props} />
+              }
+            // fallsthrough
+            case Status.Updating:
+              if (typeof Updating !== 'undefined') {
+                return (
+                  <Updating {...afterQuery(data)} {...queryRest} {...props} />
+                )
+              }
+            // fallsthrough
+            case Status.Success:
```

Comment:
I tried to make it so that this section is just concerned with rendering the right component, not figuring out `status` at the same time.

Reply:
Scratched the fall through for now.